tests/CSharp/CSharp.cs

```diff
@@ -35,6 +37,10 @@ public override void Preprocess(Driver driver, ASTContext ctx)
             ctx.SetClassAsValueType("QRect");
             ctx.SetClassAsValueType("StructTestArrayTypeFromTypedef");
             ctx.IgnoreClassWithName("IgnoredTypeInheritingNonIgnoredWithNoEmptyCtor");
+
+            var macroRegex = new Regex(@"(MY_MACRO_TEST_.*)");
+            var enumTest = ctx.GenerateEnumFromMacros("MyMacroTestEnum", (from unit in ctx.TranslationUnits where unit.FilePath == "<invalid>" || unit.FileName == "CSharp.h" from macro in unit.PreprocessedEntities.OfType<MacroDefinition>() let match = macroRegex.Match(macro.Name) where match.Success select macro.Name).ToArray());
```

Comment:
Please move the LINQ to a new line and split it into more lines, one per statement.